x-pack/plugins/cases/public/components/user_actions/property_actions/alert_property_actions.test.tsx

```diff
@@ -16,84 +16,97 @@ import {
   createAppMockRenderer,
 } from '../../../common/mock';
 import { AlertPropertyActions } from './alert_property_actions';
+import { useDeletePropertyAction } from './use_delete_property_action';
+
+jest.mock('./use_delete_property_action');
+
+for (let i = 0; i < 200; i++) {
+  describe('AlertPropertyActions', () => {
+    let appMock: AppMockRenderer;
+    const useDeletePropertyActionMock = useDeletePropertyAction as jest.Mock;
+    const onConfirmMock = jest.fn();
+
+    const props = {
+      isLoading: false,
+      totalAlerts: 1,
+      onDelete: jest.fn(),
+    };
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      appMock = createAppMockRenderer();
+      useDeletePropertyActionMock.mockReturnValue({
```

Comment:
I am skeptical about mocking this hook. What is the reasoning behind it?

Reply:
I saw that all 3 components are using this hook and according to [our findings](https://docs.google.com/document/d/1BRg4ABpSUN8JRFplePiAd6oiTqLS-nRREHsT4pckXfc/edit#heading=h.2qecycu61g7d) this could be the reason for failure.

Reply:
Are you referring to "Explicitly mock all hooks used by the component"? It is a mistake on our part. It should be "Explicitly mock all hooks that do external API calls or use React Query used by the component.". Hooks that expose logic without making any API call or use React Query (like the `useDeletePropertyAction`) should not be mocked IMO. The reason is that we need to test how they interact with our components. Otherwise, if an integration bug is introduced in the `useDeletePropertyAction` we will never catch it in our tests.

Reply:
Got it. This makes sense 👍